core/src/test/scala/codesearch/core/regex/lexer/TokenizerSpec.scala

```diff
@@ -131,22 +131,26 @@ class TokenizerSpec extends FreeSpec with Matchers {
     }
 
     "Character sets" - {
-      testParseAndRender("[^a-z]", Seq(Other("[^a-z]")))
+      testParseAndRender("[^a-z]", Seq(CharSet("[^a-z]")))
       // Parsed as set('[')
-      testParseAndRender("[[]", Seq(Other("[[]")))
+      testParseAndRender("[[]", Seq(CharSet("[[]")))
       // Parsed as set(']')
-      testParseAndRender("[]]", Seq(Other("[]]")))
+      testParseAndRender("[]]", Seq(CharSet("[]]")))
       // Parsed as set('[a]')
-      testParseAndRender("[a]]", Seq(Other("[a]"), SpecialSymbol("]")))
+      testParseAndRender("[a]]", Seq(CharSet("[a]"), SpecialSymbol("]")))
       // Parsed as set('[', ']')
-      testParseAndRender("[][]", Seq(Other("[][]")))
+      testParseAndRender("[][]", Seq(CharSet("[][]")))
       // Parsed as two sets
-      testParseAndRender("[][] [][]", Seq(Other("[][]"), SpecialSymbol(" "), Other("[][]")))
+      testParseAndRender("[][] [][]", Seq(CharSet("[][]"), SpecialSymbol(" "), CharSet("[][]")))
       // Parsed as two sets
-      testParseAndRender("[[] []]", Seq(Other("[[]"), SpecialSymbol(" "), Other("[]]")))
-      testParseAndRender("[\\]\\\\]", Seq(SpecialSymbol("["), Escaped(']'), Escaped('\\'), SpecialSymbol("]")))
-      testParseAndRender("[{3,5}]", Seq(Other("[{3,5}]")))
-      testParseAndRender("[{3,5]}", Seq(Other("[{3,5]"), SpecialSymbol("}")))
+      testParseAndRender("[[] []]", Seq(CharSet("[[]"), SpecialSymbol(" "), CharSet("[]]")))
+      testParseAndRender("[\\]\\\\]", Seq(CharSet("[\\]\\\\]")))
+      testParseAndRender("[{3,5}]", Seq(CharSet("[{3,5}]")))
+      testParseAndRender("[{3,5]}", Seq(CharSet("[{3,5]"), SpecialSymbol("}")))
+      testParseAndRender("[\\foo]", Seq(CharSet("[\\foo]")))
+      testParseAndRender("[a-z\\\\n\\\\t]", Seq(CharSet("[a-z\\\\n\\\\t]")))
```

Comment:
Sorry, I meant ordinary escaped newline+tab, not escaped slashes.

```suggestion
      testParseAndRender("[a-z\\n\\t]", Seq(CharSet("[a-z\\n\\t]")))
```